components/automate-cli/pkg/verifyserver/services/externalpostgresqlservice/externalpostgresqlservice.go

```diff
@@ -0,0 +1,99 @@
+package externalpostgresqlservice
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/db"
+	"github.com/chef/automate/lib/logger"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+)
+
+type ISExternalPostgresqlService interface {
+	GetPgConnection(*models.ExternalPgRequest) (*models.ExternalPgResponse, error)
+}
+
+type ExternalPostgresqlService struct {
+	logger  logger.Logger
+	DBUtils db.DB
+	Req     *models.ExternalPgRequest
+}
+
+func NewExternalPostgresqlService(db db.DB, logger logger.Logger) ISExternalPostgresqlService {
+	return &ExternalPostgresqlService{
+		logger:  logger,
+		DBUtils: db,
+	}
+}
+
+func (pg *ExternalPostgresqlService) GetPgConnection(req *models.ExternalPgRequest) (*models.ExternalPgResponse, error) {
+	pg.Req = req
+	err := pg.CheckExternalPgConnection()
+	if err != nil {
+		pg.logger.Error(err)
+		return &models.ExternalPgResponse{
+			Passed: false,
+			Checks: []models.ExternalPgConnectionDetails{FailResponse(ExternalPgFailConnectionTitle, ExternalPgConnectionErrorMsg, ExternalPgConnectionResolutionMsg)},
+		}, err
+	}
+	return &models.ExternalPgResponse{
+		Passed: true,
+		Checks: []models.ExternalPgConnectionDetails{SuccessResponse(ExternalPgSuccessConnectionTitle, ExternalPgConnectionSuccessMsg)},
+	}, nil
+}
+
+func (p *ExternalPostgresqlService) CheckExternalPgConnection() error {
+
+	sslrootca := p.Req.PostgresqlRootCert
+
+	tempFile, err := ioutil.TempFile("", "root-cert") // nosemgrep
+	if err != nil {
+		fmt.Println("Failed while creating a file : ", err)
+		return errors.Wrap(err, "Failed while creating a file")
+	}
+	defer os.Remove(tempFile.Name())
+	// delete the file when done
+	_, err = tempFile.WriteString((sslrootca))
+	if err != nil {
+		fmt.Println("Failed while writing rootca : ", err)
+		return errors.Wrap(err, "Failed while writing rootca")
+	}
+	connStr := fmt.Sprintf("host=%s port=%s user=%s "+
+		"password=%s dbname=postgres sslmode=verify-ca sslrootcert=%s",
+		p.Req.PostgresqlInstanceUrl, p.Req.PostgresqlInstancePort, "%s", "%s", tempFile.Name())
+	superusrconnstr := fmt.Sprintf(connStr, p.Req.PostgresqlSuperUserUserName, p.Req.PostgresqlSuperUserPassword)
```

Comment:
Please check naming convention is followed for all variables

Reply:
done